infra/modules/gcp-host/main.tf

```diff
@@ -71,10 +76,20 @@ module "secrets" {
 resource "google_secret_manager_secret_iam_member" "grant_sa_updater_on_lockable_secrets" {
   for_each = { for secret in local.lockable_secrets : secret.instance_secret_id => secret }
 
+  project   = var.gcp_project_id
+  secret_id = "${local.config_parameter_prefix}${each.value.instance_secret_id}"
   member    = "serviceAccount:${google_service_account.api_connectors[each.value.instance_id].email}"
   role      = module.psoxy.psoxy_instance_secret_locker_role_id
+
+}
+
+resource "google_secret_manager_secret_iam_member" "grant_sa_secretVersionAdder_on_writable_secret" {
+  for_each = { for secret in local.writable_secrets : secret.instance_secret_id => secret }
+
   project   = var.gcp_project_id
   secret_id = "${local.config_parameter_prefix}${each.value.instance_secret_id}"
+  member    = "serviceAccount:${google_service_account.api_connectors[each.value.instance_id].email}"
```

Comment:
Obviously it doesn't matter for TF but kind of confusing reading the code that the api connectors SA are defined below. I'd probably do:
1. define SA connectors
2. define Secrets (dependent on 1)
3. generate each api connector (dependent on 1 and 2)

right now code does 2,1,3